Fix: Load layer from URL when creating new session

Metroseksuaali opened the pull request:
## Summary
Fixes the issue where opening a URL with both `layer` and `session` parameters doesn't load the layer.

## Problem
When opening `?map=Gorodok&layer=RAASv2&session=test1234`, the layer was ignored because `loadLayers()` explicitly skips layer loading when a session parameter exists.

## Solution
Instead of removing the `!session` check (which would break joining existing sessions), I added layer loading logic to the `SESSION_CREATED` handler.

This way:
- **New session:** Layer loads from URL after session is created
- **Joining existing session:** Session state is preserved (flags, factions, units not overwritten)

## Testing
Tested locally with both scenarios:
- ✅ New session with layer in URL → layer loads correctly
- ✅ Joining existing session → session state preserved

sh4rkman commented:
Another problem : if layer is now loaded & sent to the session after session is created, the default teams are not sent. Joining the sessions will load empty teams

![20251129_11h01m_iUHGmHQimmN](https://github.com/user-attachments/assets/d90a86a3-3126-4183-b259-95df8385715e)

The session should be ideally opened after everything else is ran : 
- map is loaded from URL (if it exist)
- layer is loaded from URL (if it exist)
- session can now be opened with full data about map/layer/factions/units

it should also work for every cases, aka opening a link with no layer attached : http://test/?map=Anvil&session=etkyGa

Metroseksuaali commented:
Okay so i tried something different.
The previous approach of loading the layer in SESSION_CREATED had a timing issue - the layer was loaded AFTER the session was created, so teams/factions weren't included in the initial session state.
New approach: Instead of checking if the URL has a session parameter, I check if this.session actually exists.
Changed in squadCalc.js:
javascript// Before:
`if (currentUrl.searchParams.has("layer") && !currentUrl.searchParams.has("session"))`

// After:
`if (currentUrl.searchParams.has("layer") && !this.session)`
Why this works:

Page loads with ?map=Gorodok&layer=RAASv2&session=newtest123
First loadLayers() call: this.session is still false → layer loads from URL ✓
Session gets created (now includes the layer data)
SESSION_JOINED triggers map change → loadLayers() called again
Second call: this.session is now true → layer does NOT overwrite session state ✓

Changes in squadCalcAPI.js:
Session creation now waits for layer to fully load before instantiating SquadSession:
```
javascriptif (urlParams.has("layer")) {
    $(document).one("layer:loaded", () => {
        App.session = new SquadSession(sessionId);
    });
} else {
    $(document).one("layers:loaded", () => {
        App.session = new SquadSession(sessionId);
    });
}
```
Changes in squadSession.js:
Removed the SESSION_CREATED layer loading code from my previous attempt - it's no longer needed.

Hopefully there is nothing missing now.
